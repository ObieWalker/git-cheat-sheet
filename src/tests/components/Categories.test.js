import React from 'react';
import ReactDOM from 'react-dom';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import connect, {Categories} from '../../components/Categories';

const mockStore = configureStore();
let mounted;
let props;
let wrapper;
const categories = [
  {name:'sddsf'}, {name: 'sdfdss'}
]
const search = ''
const deleteCategory = jest.fn(() => Promise.resolve({}));
const onShow = jest.fn(() => Promise.resolve({}));
const cheatModalShow = jest.fn(() => Promise.resolve({}));
const onCopy = jest.fn(() => Promise.resolve({}));
const deleteCheat = jest.fn(() => Promise.resolve({}));
const userId = '1234'
const categoryId = '12345'

const getComponent = () => {
  if (!mounted) {
    props = {
      categories,
      search,
      deleteCategory,
      onShow,
      cheatModalShow,
      onCopy,
      deleteCheat,
      userId,
      categoryId
    };
    mounted = shallow(<Categories {...props} />);
  }
  return mounted;
};

describe('cheats Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders component successfully', () => {
    wrapper = getComponent();
    wrapper.setProps({ categories: [] });
  });


});

describe('Connected Component', () => {
  describe('Connected mounted', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        categories
      });
      wrapper = shallow(<connect
        store={store}
        categories={categories}
      />);
      expect(wrapper.length).toBe(1);
    });
  });
});


