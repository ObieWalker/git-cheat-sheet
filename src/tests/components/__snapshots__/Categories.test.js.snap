// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cheats Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Categories
    categories={
      Array [
        Object {
          "name": "sddsf",
        },
        Object {
          "name": "sdfdss",
        },
      ]
    }
    categoryId="12345"
    cheatModalShow={[MockFunction]}
    deleteCategory={[MockFunction]}
    deleteCheat={[MockFunction]}
    onCopy={[MockFunction]}
    onShow={[MockFunction]}
    search=""
    userId="1234"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CategoryCard
          cheatModalShow={[MockFunction]}
          cheats={
            Object {
              "name": "sddsf",
            }
          }
          deleteCategory={[MockFunction]}
          deleteCheat={[MockFunction]}
          onCopy={[MockFunction]}
          onShow={[MockFunction]}
          search=""
          userId="1234"
        />,
        <CategoryCard
          cheatModalShow={[MockFunction]}
          cheats={
            Object {
              "name": "sdfdss",
            }
          }
          deleteCategory={[MockFunction]}
          deleteCheat={[MockFunction]}
          onCopy={[MockFunction]}
          onShow={[MockFunction]}
          search=""
          userId="1234"
        />,
      ],
      "className": "cards-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "categoryId": undefined,
          "cheatModalShow": [MockFunction],
          "cheats": Object {
            "name": "sddsf",
          },
          "deleteCategory": [MockFunction],
          "deleteCheat": [MockFunction],
          "onCopy": [MockFunction],
          "onShow": [MockFunction],
          "search": "",
          "userId": "1234",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "categoryId": undefined,
          "cheatModalShow": [MockFunction],
          "cheats": Object {
            "name": "sdfdss",
          },
          "deleteCategory": [MockFunction],
          "deleteCheat": [MockFunction],
          "onCopy": [MockFunction],
          "onShow": [MockFunction],
          "search": "",
          "userId": "1234",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CategoryCard
            cheatModalShow={[MockFunction]}
            cheats={
              Object {
                "name": "sddsf",
              }
            }
            deleteCategory={[MockFunction]}
            deleteCheat={[MockFunction]}
            onCopy={[MockFunction]}
            onShow={[MockFunction]}
            search=""
            userId="1234"
          />,
          <CategoryCard
            cheatModalShow={[MockFunction]}
            cheats={
              Object {
                "name": "sdfdss",
              }
            }
            deleteCategory={[MockFunction]}
            deleteCheat={[MockFunction]}
            onCopy={[MockFunction]}
            onShow={[MockFunction]}
            search=""
            userId="1234"
          />,
        ],
        "className": "cards-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "categoryId": undefined,
            "cheatModalShow": [MockFunction],
            "cheats": Object {
              "name": "sddsf",
            },
            "deleteCategory": [MockFunction],
            "deleteCheat": [MockFunction],
            "onCopy": [MockFunction],
            "onShow": [MockFunction],
            "search": "",
            "userId": "1234",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "categoryId": undefined,
            "cheatModalShow": [MockFunction],
            "cheats": Object {
              "name": "sdfdss",
            },
            "deleteCategory": [MockFunction],
            "deleteCheat": [MockFunction],
            "onCopy": [MockFunction],
            "onShow": [MockFunction],
            "search": "",
            "userId": "1234",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
